### ISSUE TITLE
Incorrect method generated for routes with same URL

### ISSUE THREAD

**Issue by Shaglock** [ElMassimo/js_from_routes#48]
[troubleshooting section]: https://js-from-routes.netlify.app/faqs/

- [x] I have tried upgrading by running `bundle update js_from_routes`.
- [x] I have read the __[troubleshooting section]__ before opening an issue.

### Description 📖

I am using `devise` and `omniauth` gems (versions 4.9.4 and 2.1.2) in brand new rails app, I have 3 providers - google, twitter and facebook. The generated paths have incorrect methods `get|post`

`config/routes.rb`
```ruby
defaults export: true do
  # All routes defined inside this block will be exported.
  devise_for :users, controllers: {
    omniauth_callbacks: "users/omniauth_callbacks"
  }
end
 ```
  Output of related `rails routes`
 ```
   user_google_oauth2_omniauth_authorize GET|POST /users/auth/google_oauth2(.:format)                users/omniauth_callbacks#passthru
    user_google_oauth2_omniauth_callback GET|POST /users/auth/google_oauth2/callback(.:format)       users/omniauth_callbacks#google_oauth2
         user_twitter_omniauth_authorize GET|POST /users/auth/twitter(.:format)                      users/omniauth_callbacks#passthru
          user_twitter_omniauth_callback GET|POST /users/auth/twitter/callback(.:format)             users/omniauth_callbacks#twitter
        user_facebook_omniauth_authorize GET|POST /users/auth/facebook(.:format)                     users/omniauth_callbacks#passthru
         user_facebook_omniauth_callback GET|POST /users/auth/facebook/callback(.:format)            users/omniauth_callbacks#facebook
```         
Generated result
```ts
export default {
  passthru: /* #__PURE__ */ definePathHelper("get|post", "/users/auth/google_oauth2"),
  googleOauth2: /* #__PURE__ */ definePathHelper("get|post", "/users/auth/google_oauth2/callback"),
  twitter: /* #__PURE__ */ definePathHelper("get|post", "/users/auth/twitter/callback"),
  facebook: /* #__PURE__ */ definePathHelper("get|post", "/users/auth/facebook/callback"),
};
```

Probably its not related to devise but to the way routes are defined and how this gem handles it. 
Related code from devise gem https://github.com/heartcombo/devise/blob/72884642f5700439cc96ac560ee19a44af5a2d45/lib/devise/rails/routes.rb#L446C9-L457C1

Basically if the different HTTP methods are mapped to the same controller action this issue will appear.
For example, 
```rb
match "/test", to: "home#test", via: [ :get, :post, :patch, :put, :delete ]
```
results in 
```ts
test: /* #__PURE__ */ definePathHelper("get|post|patch|put|delete", "/test"),
```

So my question is, is there anything we can do with it? Maybe the gem could use the `as` option that was added to the route and add method as suffix?
So the result would be something like 
`google_oauth2_omniauth_authorize_get`, `google_oauth2_omniauth_authorize_post`, or in case of `test` that would be `testGet`, `testPost`, `testPatch` etc...

Or maybe there is some option for different exporting for such routes that I missed? 

In any case, thanks for the useful gem!

**Comment by Shaglock** [comment]
I just realized there was already similar issue in github "Discussions" https://github.com/ElMassimo/js_from_routes/discussions/37, I was searching only issues :( Sorry

**Comment by ElMassimo** [comment]
As a temporary workaround, you could try:

```ruby
# config/initializers/js_from_routes.rb

if Rails.env.development?
  # Internal: Override to prefer POST requests for dual paths.
  JsFromRoutes::Route.prepend Module.new {
    def verb
      @route.verb.split("|").last.downcase
    end
  }
end
```

---

I'd like to extend support to routes that are not following the default routing patterns, but I haven't decided how that should be exposed to the frontend.

Grouping routes by controller felt like a natural way to use them, while also allowing tree-shaking.

I'm not sure what would be a good convention when a route doesn't "belong" to any resource. The main problem with non-REST routes is that I haven't needed that in any of the apps I've worked on, and it's difficult to think about it without a concrete use case.
